### PR TITLE
Don't attempt to rename ZIPs coming from a GitHub archive release/tag

### PR DESCRIPTION
**Pull request by ryotsun**
Fixed Issue https://github.com/wp-cli/extension-command/issues/35

I've tried fixing against this issue.
Please review PR and give me some advice.

Thank you.

### CODE REVIEW

File: src/WP_CLI/CommandWithUpgrade.php [hunk 1]
```diff
@@ -152,15 +152,22 @@ public function install( $args, $assoc_args ) {
...
+							array_pop( $path_array );
+							$branch_name = end( $path_array );
+						}
+
+						$new_path = str_replace( pathinfo( $source, PATHINFO_BASENAME ), $branch_name, $source );
+						if ( $GLOBALS['wp_filesystem']->move( $source, $new_path ) ) {
+							WP_CLI::log( "Renamed Github-based project from '" . Utils\basename( $source ) . "' to '" . Utils\basename( $new_path ) . "'." );
+							return $new_path;
+						} else {
+							return new \WP_Error( 'wpcli_install_gitub', "Couldn't move Github-based project to appropriate directory." );
```
Comment (miya0001) [inline src/WP_CLI/CommandWithUpgrade.php]: `/wpcli_install_gitub/wpcli_install_github/` :)

**Comment by ryotsun** [comment]
🤔 
failed 1 test... 
https://travis-ci.org/wp-cli/extension-command/jobs/277782249

**Comment by ryotsun** [comment]
@danielbachhuber Sure. Sorry for forgot that.
I'm working on it. 🙇

**Comment by ryotsun** [comment]
@danielbachhuber @miya0001 
I wrote a test 👍 

Btw, test fails every time on WP 3.7.11 even if there is no code changes (means in master branch).
Therefore, it might have no relevance to my changes.
Anyway, I tried testing for the following on my local env.

```
$ WP_VERSION=3.7.11 vendor/bin/behat features/plugin-update.feature 
```

but unfortunately, I encountered the same error.

**Comment by miya0001** [comment]
HI @danielbachhuber 

I realized that the version number of the 3.7 should be `3.7.11` instead of `3.7` or `3.7.22`?
https://codex.wordpress.org/WordPress_Versions

**Comment by danielbachhuber** [comment]
> I realized that the version number of the 3.7 should be 3.7.11 instead of 3.7 or 3.7.22?

Could. I'm not sure what's causing the test failure off the top of my head though.

**Comment by miya0001** [comment]
@ryotsun 
Please merge #49 to your PR if #49 will be merged.
Then tests will be passed. :)

**Comment by ryotsun** [comment]
@danielbachhuber @miya0001 
Hello, 
finally, all checks have passed... 😄 
Please check it again.

**Comment by schlessera** [comment]
@miya0001 , @ryotsun : We now have a separate GitHub organization called `wp-cli-test` to contain stuff like the above example plugin, that should represent external packages.

Just let me know what it is you need, and I will add it to that test organization. @miya0001, should I add the above plugin as-is?

**Comment by miya0001** [comment]
> should I add the above plugin as-is?

Oh, yes please! 😊

**Comment by schlessera** [comment]
Will do and update here as soon as I got the required access. The org is brand new.

**Comment by schlessera** [comment]
@miya0001, @ryotsun : Here's the plugin you can use for testing: https://github.com/wp-cli-test/generic-example-plugin

**Comment by ryotsun** [comment]
@miya0001 @schlessera 
Thank you for your assistance.
I've updated plugin's path to the following as you said:
https://github.com/wp-cli-test/generic-example-plugin

Thank you.

File: src/WP_CLI/CommandWithUpgrade.php (hunk 1 above)
Comment (danielbachhuber) [reply src/WP_CLI/CommandWithUpgrade.php]: 6fd40e49b1635c3aa05d6efb3300049439eff21d

File: src/WP_CLI/CommandWithUpgrade.php [hunk 2]
```diff
@@ -167,7 +167,7 @@ public function install( $args, $assoc_args ) {
 							WP_CLI::log( "Renamed Github-based project from '" . Utils\basename( $source ) . "' to '" . Utils\basename( $new_path ) . "'." );
 							return $new_path;
 						} else {
-							return new \WP_Error( 'wpcli_install_gitub', "Couldn't move Github-based project to appropriate directory." );
+							return new \WP_Error( 'wpcli_install_github', "Couldn't move Github-based project to appropriate directory." );
```
Comment (ryotsun) [inline src/WP_CLI/CommandWithUpgrade.php]: Ohhh, thanks! 😢 
https://github.com/wp-cli/extension-command/blob/master/src/WP_CLI/CommandWithUpgrade.php#L162

File: src/WP_CLI/CommandWithUpgrade.php (hunk 1 above)
Comment (ryotsun) [reply src/WP_CLI/CommandWithUpgrade.php]: Sorry I couldn't catch this meanings... replace... :(

**Comment by danielbachhuber** [comment]
Thanks for your work on this @ryotsun !

**Pull request merged**